crux-core/src/crux/kv/index_store.clj

```diff
@@ -654,12 +656,13 @@
 
   (decode-value [this value-buffer]
     (assert (some? value-buffer))
-    (if (c/can-decode-value-buffer? value-buffer)
-      (c/decode-value-buffer value-buffer)
-      (or (.get temp-hash-cache value-buffer)
-          (let [i @decode-value-iterator-delay]
-            (when (advance-iterator-to-hash-cache-value i value-buffer)
-              (some-> (kv/value i) (mem/<-nippy-buffer)))))))
+    (binding [nippy/*thaw-serializable-allowlist* #{"*"}]
```

Comment:
Most of the calls to this function won't need to do any nippy deserialisation (because a decent amount of the values being decoded now go via the `c/decode-value-buffer` branch) - in these cases, we'll be installing the binding without it being used. We're also not gaining anything from the binding being done at a higher level here because, even if it is used, it's only ever used once.

(It also happens that this is also quite a low-level function called within tight loops, so the binding at this level would have had quite a performance impact! :sweat_smile:)